safe/gui/tools/options_dialog.py

```diff
@@ -107,48 +106,89 @@ def __init__(self, iface, dock=None, parent=None, qsetting=''):
+        key = 'setHideExposureFlag'
         flag = bool(self.settings.value(
-            'inasafe/setHideExposureFlag', False, type=bool))
+            'inasafe/%s' % key, inasafe_default_settings[key], type=bool))
         self.cbxHideExposure.setChecked(flag)
 
+        key = 'useSelectedFeaturesOnly'
         flag = bool(self.settings.value(
-            'inasafe/useSelectedFeaturesOnly', False, type=bool))
+            'inasafe/%s' % key, inasafe_default_settings[key], type=bool))
         self.cbxUseSelectedFeaturesOnly.setChecked(flag)
 
+        key = 'useSentry'
         flag = bool(self.settings.value(
-            'inasafe/useSentry', False, type=bool))
+            'inasafe/%s' % key, inasafe_default_settings[key], type=bool))
         self.cbxUseSentry.setChecked(flag)
 
-        path = self.settings.value(
-            'inasafe/keywordCachePath',
-            self.keyword_io.default_keyword_db_path(), type=str)
-        self.leKeywordCachePath.setText(path)
-
+        key = 'template_warning_verbose'
         flag = bool(self.settings.value(
-            'inasafe/template_warning_verbose', True, type=bool))
+            'inasafe/%s' % key, inasafe_default_settings[key], type=bool))
         self.template_warning_checkbox.setChecked(flag)
 
+        # Restore Show Organisation Logo in Dock Flag
+        key = 'showOrganisationLogoInDockFlag'
+        flag = bool(self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key], type=bool))
+        self.organisation_on_dock_checkbox.setChecked(flag)
+
+        key = 'developer_mode'
+        flag = bool(self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key], type=bool))
+        self.cbxDevMode.setChecked(flag)
+
+        # Restore ISO19115 metadata tab
+        key = 'ISO19115_ORGANIZATION'
+        value = self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key], type=str)
+        self.iso19115_organization_le.setText(value)
+
+        key = 'ISO19115_URL'
+        value = self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key], type=str)
+        self.iso19115_url_le.setText(value)
+
+        key = 'ISO19115_EMAIL'
+        value = self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key], type=str)
+        self.iso19115_email_le.setText(value)
+
+        key = 'ISO19115_TITLE'
+        value = self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key], type=str)
+        self.iso19115_title_le.setText(value)
+
+        key = 'ISO19115_LICENSE'
+        value = self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key], type=str)
+        self.iso19115_license_le.setText(value)
+
+        # Keyword cache path
+        key = 'keywordCachePath'
+        path = self.settings.value(
+            'inasafe/%s' % key, inasafe_default_settings[key])
+        self.leKeywordCachePath.setText(path)
```

Comment:
All of the code above can be rewritten as for loops, something like this:

```
settings = [{
    'key': 'keywordCachePath',
    'control': self.leKeywordCachePath,
}]

for setting in settings:
    restoreControlSetting(setting['key'], setting['control'])
```

For me it improves readability, and reduce mistakes from copy pasting when you refactored it.
But it's up to you.